crates/loader/src/local/assets.rs

```diff
@@ -208,13 +213,25 @@ async fn copy(file: &FileMount, dir: impl AsRef<Path>) -> Result<()> {
 
     tokio::fs::create_dir_all(to.parent().expect("Cannot copy to file '/'")).await?;
 
+    // if destination file is read-only, set it to writable first
+    let metadata = tokio::fs::metadata(&to).await;
+    if metadata.is_ok() && metadata.unwrap().permissions().readonly() {
+        let mut perms = tokio::fs::metadata(&to).await?.permissions();
+        perms.set_readonly(false);
+        tokio::fs::set_permissions(&to, perms.clone())
+            .await
+            .with_context(|| anyhow!("Cannot set permission {:?}", perms.clone()))?;
+    }
+
     let _ = tokio::fs::copy(&from, &to)
         .await
         .with_context(|| anyhow!("Error copying asset file  '{}'", from.display()))?;
 
-    let perms = tokio::fs::metadata(&to).await?.permissions();
-    // perms.set_readonly(true);
-    tokio::fs::set_permissions(&to, perms).await?;
+    let mut perms = tokio::fs::metadata(&to).await?.permissions();
```

Comment:
The `let mut perms = ...; perms.set_readonly(...); set_permissions(..., perms).await.with_context(...)` sequence seems to happen in quite a few places.  Could we create a utility function (e.g. `set_writeable` or `allow_write`)  to tidy and reveal intent?

Reply:
Yes, I add `change_file_permission` function for it. Thank you.